### PR TITLE
Fix empty stringify under C++20

### LINKED ISSUE #220: Seg fault under C++20 and empty stringify parameter.
Tested on the development branch.

```C
#define t(x, ...) #x#__VA_ARGS__
int main()
{
  t(1);
}
```

Triggers a segment fault when Wave tries to stringify the empty string; the check [here](https://github.com/boostorg/wave/blob/7ce8b53d9dc56a9c0b9a8855f26003e6b174301f/include/boost/wave/util/cpp_macromap.hpp#L1266) does nothing as `i` is off-the-bound.

I think I will PR to fix this...

### PR DESCRIPTION
**Pull request by jwnhy**
Fix #220.

Check if the variadics variable is omitted; put an empty string if so.

### CODE REVIEW

**Comment by jefftrull** [comment]
Thank you for your issues and PRs, which I am going through slowly :) I sent you a test for one of them.
Will be traveling for a week so progress will be slow.

File: include/boost/wave/util/cpp_macromap.hpp [hunk 1]
```diff
@@ -1262,27 +1262,39 @@ macromap<ContextT>::expand_replacement_list(
...
-                        impl::as_stringlit(arguments, i, pos), pos));
-                }
+#if BOOST_WAVE_SUPPORT_CPP2A != 0
+                if (i >= arguments.size()) {
+                    // no argument supplied; do nothing (only c20 should reach here)
+                    BOOST_ASSERT(boost::wave::need_cpp2a(ctx.get_language()));
+                    position_type last_valid(arguments.back().back().get_position());
+                    // insert a empty string
+                    expanded.push_back(token_type(T_STRINGLIT, "\"\"", last_valid));
+                    adjacent_stringize = false;
```
Comment (jefftrull) [inline include/boost/wave/util/cpp_macromap.hpp:1272]: It seems to me that you could leave this out and let the code fall through to the original `adjacent_stringize = false;`

File: include/boost/wave/util/cpp_macromap.hpp [hunk 2]
```diff
@@ -1262,27 +1262,39 @@ macromap<ContextT>::expand_replacement_list(
...
+                            impl::as_stringlit(arguments, i, pos), pos));
+                    }
+                    else
+#endif
+                    {
+                        impl::trim_sequence(arguments[i]);
+                        expanded.push_back(token_type(T_STRINGLIT,
+                            impl::as_stringlit(arguments[i], pos), pos));
+                    }
+                    adjacent_stringize = false;
```
Comment (jefftrull) [inline include/boost/wave/util/cpp_macromap.hpp:1296]: and this one

File: include/boost/wave/util/cpp_macromap.hpp [hunk 3]
```diff
@@ -1262,27 +1262,39 @@ macromap<ContextT>::expand_replacement_list(
...
                 }
-                adjacent_stringize = false;
```
Comment (jefftrull) [inline include/boost/wave/util/cpp_macromap.hpp:1285]: and put this one back in? Three fewer changed lines. Otherwise I'm happy with it.

**Comment by jwnhy** [comment]
Thank you!
Have just pushed a fix to put the `adjacent_stringize = false;` back in place.

**Comment by jefftrull** [comment]
Thank you for this contribution.

**Pull request merged**